lib/src/clojure_lsp/feature/completion.clj

```diff
@@ -450,8 +456,14 @@
       item)))
 
 (defn resolve-item [{{:keys [ns]} :data :as item} db]
-  (if (:data item)
-    (if ns
-      (resolve-item-by-ns item db)
-      (resolve-item-by-definition item db))
-    item))
+  (let [item (shared/assoc-some item
+                                :insert-text-format (:insertTextFormat item)
+                                :text-edit (:textEdit item)
+                                :filter-text (:filterText item)
+                                :additional-text-edits (:additionalTextEdits item)
+                                :insert-text (:insertText item))]
```

Comment:
I think is not that simple, as for example `:additionalTextEdits` will return a List<TextEdit>, not a clojure list of maps representing the text edit, right?
This will probably fail the coercer which expect clojure maps and lists but will get now a bunch of fields which are classes.

Maybe the best is to try to make the borders parse from/to java/clojure properly so the feature/completion always handle keywords, do you think that would be possible?

Reply:
I didn't about `:additionalTextEdits`, good catch.

Yeah this whole thing is kind of ugly, but I couldn't find a good way handle camelCase <-> kebab-case with the way `lsp4j` flows into the `clojure.java.data` functions. I'd like to do the camelCase <-> kebab-case conversion by hand in the `clojure.java.data/from-java` multimethod, but `clojure-lsp.server/sync-request` pipes the `clojure-lsp.coercer/java->clj` call directly into the handler. 
I think the cleanest solution would be to also pipe this through a request-spec spec before calling the handler. That way we can change the casing or any other stuff that might come up at the "right" location.

What do you think?

Reply:
Yes, sounds better indeed